did_resolver_sov/src/resolution/utils.rs

```diff
@@ -0,0 +1,181 @@
+use chrono::{DateTime, NaiveDateTime, Utc};
+use did_resolver::{
+    did_doc_builder::schema::{did_doc::DIDDocument, service::Service, types::uri::Uri},
+    did_parser::ParsedDID,
+    shared_types::did_document_metadata::DIDDocumentMetadata,
+    traits::resolvable::{
+        resolution_metadata::DIDResolutionMetadata, resolution_output::DIDResolutionOutput,
+    },
+};
+use serde_json::Value;
+
+use crate::{
+    error::{parsing::ParsingErrorSource, DIDSovError},
+    service::{DidSovServiceType, EndpointDidSov},
+};
+
+fn prepare_ids(did: &str) -> Result<(Uri, ParsedDID), DIDSovError> {
+    let service_id = Uri::new(did)?;
+    let ddo_id = ParsedDID::parse(did.to_string())?;
+    Ok((service_id, ddo_id))
+}
+
+fn get_data_from_response(resp: &str) -> Result<Value, DIDSovError> {
+    let resp: serde_json::Value = serde_json::from_str(resp)?;
+    match &resp["result"]["data"] {
+        Value::String(ref data) => serde_json::from_str(data).map_err(|err| err.into()),
+        Value::Null => Err(DIDSovError::NotFound("DID not found".to_string())),
+        resp => Err(DIDSovError::ParsingError(
+            ParsingErrorSource::LedgerResponseParsingError(format!(
+                "Unexpected data format in ledger response: {resp}"
+            )),
+        )),
+    }
+}
+
+fn get_txn_time_from_response(resp: &str) -> Result<i64, DIDSovError> {
+    let resp: serde_json::Value = serde_json::from_str(resp)?;
+    let txn_time = resp["result"]["txnTime"]
+        .as_i64()
+        .ok_or(DIDSovError::ParsingError(
+            ParsingErrorSource::LedgerResponseParsingError("Failed to parse txnTime".to_string()),
+        ))?;
+    Ok(txn_time)
+}
+
+fn posix_to_datetime(posix_timestamp: i64) -> Option<DateTime<Utc>> {
+    NaiveDateTime::from_timestamp_opt(posix_timestamp, 0)
+        .map(|date_time| DateTime::<Utc>::from_utc(date_time, Utc))
+}
+
+pub(super) fn is_valid_sovrin_did_id(id: &str) -> bool {
+    if id.len() < 21 || id.len() > 22 {
+        return false;
+    }
+    for c in id.chars() {
+        match c {
+            '1'..='9' | 'A'..='H' | 'J'..='N' | 'P'..='Z' | 'a'..='k' | 'm'..='z' => {}
+            _ => return false,
+        }
+    }
+    true
+}
+
+pub(super) async fn resolve_ddo(did: &str, resp: &str) -> Result<DIDResolutionOutput, DIDSovError> {
```

Comment:
feel like `map_ledger_response_to_ddo` would be more appropriate, as for function doing resolution I would expect it to read the ledger as well, but that's injected

Reply:
Good point, renamed to `ledger_response_to_ddo`.